libs/datamodel/connectors/dml/src/relation_info.rs

```diff
@@ -33,23 +38,58 @@ impl RelationInfo {
             fields: Vec::new(),
             references: Vec::new(),
             name: String::new(),
-            on_delete: OnDeleteStrategy::None,
+            on_delete: None,
+            on_update: None,
+            legacy_referential_actions: false,
         }
     }
+
+    /// Set referential action legacy mode, skipping the validation errors on
+    /// automatically set actions.
+    pub fn legacy_referential_actions(&mut self) {
+        self.legacy_referential_actions = true;
+    }
 }
 
 /// Describes what happens when related nodes are deleted.
+#[repr(u8)]
+#[bitflags]
 #[derive(Debug, Copy, PartialEq, Clone)]
-pub enum OnDeleteStrategy {
+pub enum ReferentialAction {
+    /// Deletes record if dependent record is deleted. Updates relation scalar
+    /// fields if referenced scalar fields of the dependent record are updated.
+    /// Prevents operation (both updates and deletes) from succeeding if any
+    /// records are connected. This behavior will always result in a runtime
+    /// error for required relations.
```

Comment:
I do not understand the last two sentences in this context. I think they apply to `Restrict`.

Reply:
copy&pasta, fixing